### PR TITLE
test: Add tests for different postgresql scenarios 

### PR DESCRIPTION
**Pull request by k3yss**
This PR modularizes the postgresql terraform module and adds tests for three conditions:
1: Normal database working
2: Postgresql database being a source db for database migration service
3: Postgresql database being a destination db for database migration service

### CODE REVIEW

File: examples/gcp/postgresql/main.tf [hunk 1]
```diff
@@ -16,5 +16,37 @@ module "postgresql" {
   databases              = ["test"]
   replication            = true
   provision_read_replica = true
-  upgradable             = true
+  source_db_upgradable   = false
```
Comment (bodymindarts) [inline examples/gcp/postgresql/main.tf:19]: isn't this the default?

File: modules/postgresql/gcp/variables.tf [hunk 2]
```diff
@@ -45,8 +45,15 @@ variable "provision_read_replica" {
 variable "big_query_connection_location" {
   default = "US"
 }
-variable "upgradable" {
-  description = "Instance is upgradable via Database Migration Service"
+
+variable "source_db_upgradable" {
```
Comment (bodymindarts) [inline modules/postgresql/gcp/variables.tf:49]: Something is weird with the `_upgradeable` postfix. how about:

`prep_upgrade_as_source_db`?

File: modules/postgresql/gcp/variables.tf [hunk 3]
```diff
@@ -45,8 +45,15 @@ variable "provision_read_replica" {
...
+  description = "Configure source destination instance to be upgradable via Database Migration Service"
+  type        = bool
+  default     = false
+}
+
+variable "destination_db_upgradable" {
```
Comment (bodymindarts) [inline modules/postgresql/gcp/variables.tf:55]: How about:
`prep_upgrade_as_destination_db`?

File: examples/gcp/postgresql/main.tf (hunk 1 above)
Comment (k3yss) [reply examples/gcp/postgresql/main.tf]: Yes, simplified it in the latest commit

File: examples/gcp/Makefile [hunk 4]
```diff
@@ -46,7 +46,7 @@ destroy-bootstrap:
 destroy-postgresql:
 	cd postgresql \
 		&& $(TF) destroy || true \
-		&& $(TF) state rm module.postgresql.google_sql_user.admin || true \
+		&& $(TF) state rm module.postgresql_source.google_sql_user.admin || true \
```
Comment (bodymindarts) [inline examples/gcp/Makefile:49]: is this correct? Shouldn't it be `postgresql_migration_source`?

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**